image-layout.md

```diff
@@ -156,4 +161,44 @@ $ cat ./blobs/sha256/e692418e4cbaf90ca69d05a66403747baa33ee08806650b51fab815ad7f
 [tar stream]
 ```
 
+## index.json
+
+This OPTIONAL file is for use indexing references and elements of the image-layout, where walking the `refs` is sub-optimal.
+It is intended to be generated from the content in an image-layout after any changes.
+
+The [manifest-list](manifest-list.md) is an multi-descriptor entry point.
+This index provides an established path (`/index.json`) to have an entry point for this image-layout and discover auxiliary descriptors pertaining to the references.
+
+### Index Example
+
+```json,title=Manifest%20List&mediatype=application/vnd.oci.image.manifest.list.v1%2Bjson
+{
+  "schemaVersion": 2,
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "size": 7143,
+      "digest": "sha256:e692418e4cbaf90ca69d05a66403747baa33ee08806650b51fab815ad7fc331f",
+      "annotations": {
+        "org.opencontainers.ref.name": "v1.0"
```

Comment:
> walking the reference dir and using the filename as the reference name worked but if this annotation isn't provided, what should implementos use as "the reference name"?

Walking the reference directory [didn't work over bare HTTP][1], and we're likely [dropping `refs/`][2] anyway.  I still don't think this annotation needs to be required though.  If it's not set, the entry just doesn't have a “reference name”.  You may still be able to address it through other means (e.g. maybe it's the only entry with a given media type, or it has some other annotation that you can use to identify it).

[1]: https://github.com/opencontainers/image-spec/pull/438#issue-187078763
[2]: https://github.com/opencontainers/image-spec/pull/533#issuecomment-276618123